Packs/Cryptosim/Integrations/Cryptosim/Cryptosim.py

```diff
@@ -0,0 +1,234 @@
+
+    return CommandResults(
+        outputs_prefix='CorrelationAlerts',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    :type client: ``Client``
+    :param Client: client to use
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        if client.correlations().get('StatusCode') == 200:
+            message = 'ok'
+        else:
+            raise Exception(f"""StatusCode: 
+                            {client.correlations().get('StatusCode')}, 
+                            Error: {client.correlations().get('ErrorMessage')}
+                            """)
+    except DemistoException as e:
+        if '401' in str(e):
+            message = 'Authorization Error: make sure API User and Password is correctly set'
+        else:
+            raise e
+    return message
+
+
+''' INCIDENT '''
+
+
+def fetch_incidents(client: Client):
+
+    max_results = arg_to_number(arg=demisto.params().get('max_fetch'), arg_name='max_fetch', required=False)
+
+    first_fetch_time = arg_to_datetime(demisto.params().get('first_fetch')).isoformat()
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', first_fetch_time)
+
+    incidentsList = []
+    alert_response = client.correlation_alerts(start_time=last_fetch)
+    incident_data = alert_response['Data']
+
+    for inc in incident_data:
+
+        if len(incidentsList) > max_results:
+            break
+
+        incident_name = inc['CorrelationAlert']['NAME']
+        time_stamp = inc['CorrelationAlert']['CREATEDATE'] + "Z"
+
+        severity_level = int(inc['CorrelationAlert']['RISK'])
+        if severity_level >= 0 and severity_level <= 5:
+            severity = 1
+        elif severity_level > 5 and severity_level <= 7:
+            severity = 2
+        elif severity_level > 7 and severity_level <= 9:
+            severity = 3
+        elif severity_level > 9 and severity_level <= 10:
+            severity = 4
+        else:
+            severity = 0
+
+        # "log" column is stringfyed 'Log' data.
+        inc['Log'].pop("log")
+
+        incident_object = {**inc['Log'], **inc['CorrelationAlert']}
+
+        incident = {
+            'name': incident_name,
+            'occurred': time_stamp,
+            'rawJSON': json.dumps(incident_object),
+            "severity": severity,
+            'type': 'Crpyotsim CorrelationAlert'
+        }
+
+        incidentsList.append(incident)
+
+        created_incident = datetime.strptime(time_stamp, DATE_FORMAT)
+        last_fetch = datetime.strptime(last_fetch, DATE_FORMAT)
+        if created_incident > last_fetch:
+            last_fetch = created_incident
```

Comment:
Duplicates must be avoided, so take the time of the next incident which you have not yet entered into the list.

Reply:
Each alert is unique and must be converted to incident so we cannot control with time. I don't understand what you mean for that.

Reply:
for example: 
start = 15:00
max_fetch = 20
end = now = 17:00
We got from the API 25 alerts.
So when we will get to the 20th we are getting its `created_incident` to the `last_fetch`.
So the next fetch we will get again this 20th alert, instead of the 21th

Reply:
I hope it is much clearer, if not we can zoom